Source/Core/Core/Movie.cpp

```diff
@@ -854,9 +854,9 @@ void RecordInput(GCPadStatus* PadStatus, int controllerID)
 
   CheckPadStatus(PadStatus, controllerID);
 
-  EnsureTmpInputSize((size_t)(s_currentByte + 8));
-  memcpy(&(tmpInput[s_currentByte]), &s_padState, 8);
-  s_currentByte += 8;
+  EnsureTmpInputSize((size_t)(s_currentByte + sizeof(ControllerState)));
+  *reinterpret_cast<ControllerState*>(&tmpInput[s_currentByte]) = s_padState;
```

Comment:
My limited understanding is that it's okay to cast between `char*` and anything, but I'm definitely no expert. I thought this way made it more explicit what was going on, but I can change it back to the way it was before.

@lioncash you're the C++ guy, right? How's this look?